AU2/plugins/custom_plugins/PolicePlugin.py

```diff
@@ -0,0 +1,228 @@
+from AU2.plugins.CorePlugin import registered_plugin
+from AU2.plugins.constants import COLLEGES, WATER_STATUSES, WEBPAGE_WRITE_LOCATION
+from AU2.plugins.util import random_data
+from AU2.plugins.util.DeathManager import DeathManager
+from AU2.plugins.util.PoliceRankManager import DEFAULT_RANKS, PoliceRankManager
+
+POLICE_TABLE_TEMPLATE = """
+<p xmlns="">
+      Here is a list of members of our loyal police, charged with protecting Cambridge from murderers of innocents and people that annoy the all powerful Umpire:
+   </p>
+<table xmlns="" class="playerlist">
+  <tr><th>Rank</th><th>Pseudonym</th><th>Real Name</th><th>Email Address</th><th>College</th><th>Notes</th></tr>
+  {ROWS}
+</table>
+"""
+
+POLICE_TABLE_ROW_TEMPLATE = """
+<tr><td>{RANK}</td><td>{PSEUDONYM}</td><td>{NAME}</td><td>{EMAIL}</td><td>{COLLEGE}</td><td>{NOTES}</td></tr>
+"""
+
+DEAD_POLICE_TABLE_TEMPLATE = """
+<p xmlns="">
+      Killed in the line of duty:
+   </p>
+<table xmlns="" class="playerlist">
+  <tr><th>Rank</th><th>Pseudonym</th><th>Real Name</th><th>Email Address</th><th>College</th><th>Notes</th></tr>
+  {ROWS}
+</table>
+"""
+
+DEAD_POLICE_TABLE_ROW_TEMPLATE = """
+<tr><td>{RANK}</td><td>{PSEUDONYM}</td><td>{NAME}</td><td>{EMAIL}</td><td>{COLLEGE}</td><td>{NOTES}</td></tr>
+"""
+
+DEAD_CORRUPT_POLICE_TABLE_TEMPLATE = """
+<p xmlns="">
+      And these are the names of those who turned to the dark side, and paid the price:
+   </p>
+<table xmlns="" class="playerlist">
+  <tr><th>Rank</th><th>Pseudonym</th><th>Real Name</th><th>Email Address</th><th>College</th><th>Notes</th></tr>
+  {ROWS}
+</table>
+"""
+
+DEAD_CORRUPT_POLICE_TABLE_ROW_TEMPLATE = """
+<tr><td>{RANK}</td><td>{PSEUDONYM}</td><td>{NAME}</td><td>{EMAIL}</td><td>{COLLEGE}</td><td>{NOTES}</td></tr>
+"""
+
+NO_DEAD_POLICE = """<p xmlns="">No police have been killed... yet.</p>"""
+
+POLICE_PAGE_TEMPLATE: str
+with open(os.path.join(ROOT_DIR, "plugins", "custom_plugins", "html_templates", "police.html"), "r") as F:
+    POLICE_PAGE_TEMPLATE = F.read()
+
+@registered_plugin
+class PolicePlugin(AbstractPlugin):
+    def __init__(self):
+        super().__init__("PolicePlugin")
+
+        self.html_ids = {
+            "Rank": self.identifier + "_rank",
+            "Rank Name": self.identifier + "_name"
+        }
+
+        self.config_exports = [
+            ConfigExport(
+                "PolicePlugin_set_ranks",
+                "Police -> Set Ranks",
+                self.ask_set_ranks,
+                self.answer_set_ranks
+            )
+        ]
+
+        self.ranks = GENERIC_STATE_DATABASE.arb_state.setdefault(self.identifier, DEFAULT_RANKS)
+
+    def ask_set_ranks(self):  # TODO Make Ctrl-C work - not sure why it isn't
+        return [
+            Label("Rename different ranks of police to fit the game theme"),
+            Label("Ranks affect position on the police page, and can be used to reward/punish/just have a bit of fun"),
```

Comment:
I think umpires will know what's going on if they're using this plugin, so it should be safe to delete

```suggestion
```

Reply:
Personally I had no clue what police ranks were for in my game, but I guess this would make more sense in the docs/umpiring guide

Reply:
if you think it adds value you can leave them. But I imagine leading with a `Label` is what's causing your back-out failure. Let me check out to your branch and confirm.

Reply:
Why you can't go back is nothing you've done... I'll get this sorted in another branch.

Reply:
https://github.com/jsyiek/AU2/pull/27